### PR TITLE
#156005187 Handling Configuration Changes

### PR DESCRIPTION
**Pull request by codestaintin**
#### What does this PR do?
Handling Configuration Changes
#### Description of Task to be completed?
- create interface to handle configuration changes
- implement interface method in MainActivity.java
- update interface design to implement grid view
- change all List types to ArrayList for efficiency
#### How should this be manually tested?
- run `git clone` to clone this repository
- `cd` to the repository
- run `git checkout feature/156005187/handling-configuration-changes`
- run app
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
- story title - Handling Configuration Changes
- story type - feature
- story id - #156005187
#### Screenshots (if appropriate)
N/A
#### Questions
N/A

### CODE REVIEW

File: app/src/main/java/com/example/isioyemohammed/gitlist/GitListAdapter.java [hunk 1]
```diff
@@ -121,7 +109,7 @@ public ViewHolder(View itemView) {
          */
         public void setValues(@NonNull GithubUsers userList) {
             this.textViewUsername.setText(userList.getLogin());
-            this.textViewCompany.setText(userList.getUrl());
+//            this.textViewCompany.setText(userList.getUrl());
```
Comment (TheDancerCodes) [inline app/src/main/java/com/example/isioyemohammed/gitlist/GitListAdapter.java]: @codestaintin 
It's not good practise to leave commented out code in source control. If you aren't using this you should delete it.

File: app/src/main/java/com/example/isioyemohammed/gitlist/presenter/GithubUsersPresenter.java [hunk 2]
```diff
@@ -28,36 +25,48 @@
...
                 .getAPI()
                 .getItems()
                 .enqueue(new Callback<GithubUsersResponse>() {
                     @Override
                     public void onResponse(@NonNull Call<GithubUsersResponse> call,
                                            @NonNull Response<GithubUsersResponse> response) {
-                        List<GithubUsers> githubUsersResponse = response.body().getItems();
+                        GithubUsersResponse userResponse = response.body();
+                        ArrayList<GithubUsers> githubUsersResponse;
+                        assert userResponse != null;
```
Comment (TheDancerCodes) [inline app/src/main/java/com/example/isioyemohammed/gitlist/presenter/GithubUsersPresenter.java:66]: @codestaintin 
Could you please explain the purpose of this line of code? It looks like a JUnit test.

**Review by TheDancerCodes — CHANGES_REQUESTED** [review]
Hi @codestaintin,
Please look into these.
Thanks

File: app/src/main/java/com/example/isioyemohammed/gitlist/presenter/GithubUsersPresenter.java (hunk 2 above)
Comment (codestaintin) [reply app/src/main/java/com/example/isioyemohammed/gitlist/presenter/GithubUsersPresenter.java:66]: It is an alternative to and if-else statement, I chose that approach to reduce the lines of code. Is it a good practice?

**Pull request merged**